MdeModulePkg/Library/BootMaintenanceManagerUiLib/BootMaintenanceManagerCustomizedUiSupport.c

```diff
@@ -381,88 +381,90 @@ BmmListThirdPartyDrivers (
 
   HiiHandles = HiiGetHiiHandles (NULL);
   ASSERT (HiiHandles != NULL);
+  // MU_CHANGE - Ensure HiiHandles are valid before using
+  if (HiiHandles != NULL) {
+    gHiiDriverList = AllocateZeroPool (UI_HII_DRIVER_LIST_SIZE * sizeof (UI_HII_DRIVER_INSTANCE));
+    ASSERT (gHiiDriverList != NULL);
+    DriverListPtr = gHiiDriverList;
+    CurrentSize   = UI_HII_DRIVER_LIST_SIZE;
+
+    for (Index = 0, Count = 0; HiiHandles[Index] != NULL; Index++) {
+      if (!IsRequiredDriver (HiiHandles[Index], ClassGuid, &Token, &TokenHelp, &gHiiDriverList[Count].FormSetGuid)) {
+        continue;
+      }
 
-  gHiiDriverList = AllocateZeroPool (UI_HII_DRIVER_LIST_SIZE * sizeof (UI_HII_DRIVER_INSTANCE));
-  ASSERT (gHiiDriverList != NULL);
-  DriverListPtr = gHiiDriverList;
-  CurrentSize   = UI_HII_DRIVER_LIST_SIZE;
+      String = HiiGetString (HiiHandles[Index], Token, NULL);
+      if (String == NULL) {
+        String = HiiGetString (HiiHandle, STRING_TOKEN (STR_MISSING_STRING), NULL);
+        ASSERT (String != NULL);

```

Comment:
There are a couple of ways that the string would not be available:
1. the string def was removed from MdeModulePkg\Library\BootMaintenanceManagerUiLib\BootMaintenanceManagerStrings.uni
2. The HiiDatabase had the string overwritten or deleted.

Since this code is going through all the HiiHandles, and not just the Hii assocaited with this driver, it might be better to wrap the FreePool in a check if String is NULL.